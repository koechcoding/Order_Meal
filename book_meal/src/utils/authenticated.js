export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}