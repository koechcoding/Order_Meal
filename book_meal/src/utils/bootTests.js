import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
