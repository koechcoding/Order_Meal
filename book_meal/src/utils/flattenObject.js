export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}