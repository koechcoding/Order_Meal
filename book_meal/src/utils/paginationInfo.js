export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}