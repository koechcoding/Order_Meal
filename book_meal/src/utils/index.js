import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}