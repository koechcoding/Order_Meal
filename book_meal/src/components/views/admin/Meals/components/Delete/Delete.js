import React from 'react';
import PropTypes from 'prop-types';
import axios from 'src/axios';
import Modal from 'src/components/common/Modal';
import { Alert, Button } from 'reactstrap';
import { singleError } from 'src/utils';

class DeleteModal extends React.Component {
     
    state = {}

    onOpened=()=>{
        this.setState({
            ...this.state,
            error: null,
        });
    }

    onDelete=()=>{
        axios.delete('/meals/${this.props.meal.id}').then(()=>{
            this.props.onChange();
            this.props.toggle();
        }).catch(({ response})=>{
            this.setState({
                ...this.state,
                error: response,
            });
        })
    }

    render(){
        const meal = this.props.meal || {}
        const { error} = this.state;
        const bpdy = (
            <div>
              {error &&
              <Alert className="text-center text-small" color="danger">
                 { singleError(error) }  
              </Alert>
              }
              <p className='text-center'>
                Delete this meal (<b>{meal.name}</b>)?
              </p>
            </div>
        );

        const footer =(
            <Button color="danger" className="m-auto" onClick={this.onDelete}>Delete</Button>
        );

        const { isOpen, toggle } = this.props;
        return(
            <Modal
               title="Delete Meal"
               body={body}
               footer={footer}
               isOpen={isOpen}
               toggle={toggle}
               onOpened={this.onOpened}
            />
        );
    }
}

DeleteModal.propTypes = {
    meal: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired,
    toggle: PropTypes.func.isRequired,
    isOpen: PropTypes.bool.isRequired,
}

export default DeleteModal;