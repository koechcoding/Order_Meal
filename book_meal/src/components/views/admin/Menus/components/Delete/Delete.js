import React from 'react';
import PropTypes from 'prop-types';
import axios from 'src/axios';
import Modal from 'src/components/common/Modal';
import { Alert, Button } from 'reactstrap';
import { singleError } from 'src/utils';


class DeleteModal extends React.Component {
    state = {}

    onOpened =()=>{
        this.setState({
            ...this.state,
            error: null,
        });
    }

    onDelete=()=>{
        axios.delete('/menu-items/${this.props.menuItem.id}').then(()=>{
            this.props.onChange();
            this.props.toggle();
        }).catch(({ response }) =>{
            this.setState({
                ...this.state,
                error: response,
            })
        })
    }

    render(){
        const { error } = this.state;
        const { menuItem ={}} = this.props;
        const body = (
            <div>
            {error &&
                <Alert className="text-center text-small" color="danger">
                    { singleError(error) }
                </Alert>
            }
                <p className="text-center">
                    Delete this menu item (ID: <b>{menuItem.id}</b>)?
                </p>
        </div>
       );

       const footer =(
        <Button color="danger" className="m-auto" onClick={this.onDelete}>Delete</Button>
       );

       const { isOpen, toggle } = this.props;
       return (
        <Modal
           title="Delete Menu Item"
           body={body}
           footer={footer}
           isOpen={isOpen}
           toggle={toggle}
           onOpened={this.onOpened}
        />
       );
    }
}
Modal.propTypes = {
    isOpen: PropTypes.bool.isRequired,
    toggle: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    menuItem: PropTypes.object.isRequired,
}

export default DeleteModal;