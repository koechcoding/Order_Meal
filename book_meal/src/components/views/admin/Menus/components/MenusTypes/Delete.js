
import React from 'react';
import axios from 'src/axios';
import PropTypes from 'prop-types';
import { Alert, Button } from 'reactstrap';
import Modal from 'src/components/common/Modal';
import { singleError } from 'src/utils';