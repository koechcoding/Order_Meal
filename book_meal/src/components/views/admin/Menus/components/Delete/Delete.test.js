
import React from 'react';
import { shallow }  from 'enzyme';
import 'src/utils/bootTests'; 
import Delete from './Delete';
import Adapter from 'enzyme-adapter-react-16';

const props = {isOpen: false, toggle: () => {}, onChange: () => {}, menuItem: {}};
describe('<Delete />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Delete {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('resets on opened', () => {
        const wrapper = shallow(<Delete {...props }/>);
        wrapper.setState({...wrapper.state(), error: 'error'});
        wrapper.instance().onOpened();
        expect(wrapper.state('error')).toBe(null);
    });
});