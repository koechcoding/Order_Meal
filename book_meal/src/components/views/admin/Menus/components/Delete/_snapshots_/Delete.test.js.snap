// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Delete /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeleteModal
    isOpen={false}
    menuItem={Object {}}
    onChange={[Function]}
    toggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "body": <div>
        <p
                className="text-center"
        >
                Delete this menu item (ID: 
                <b />
                )?
        </p>
</div>,
      "footer": <Button
        className="m-auto"
        color="danger"
        onClick={[Function]}
        tag="button"
>
        Delete
</Button>,
      "isOpen": false,
      "onOpened": [Function],
      "title": "Delete Menu Item",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "body": <div>
          <p
                    className="text-center"
          >
                    Delete this menu item (ID: 
                    <b />
                    )?
          </p>
</div>,
        "footer": <Button
          className="m-auto"
          color="danger"
          onClick={[Function]}
          tag="button"
>
          Delete
</Button>,
        "isOpen": false,
        "onOpened": [Function],
        "title": "Delete Menu Item",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;